runtime/thunk/thunk.ts

```diff
@@ -86,14 +91,18 @@ function doFindAndExecutePackFunction<T extends FormulaSpecification>({
     }
     case FormulaType.Sync: {
       const formula = findSyncFormula(manifest, formulaSpec.formulaName);
-      return formula.execute(params, executionContext as SyncExecutionContext);
+      // for some reason TS can't tell that the return type is correctly GenericSyncFormulaResult.
+      return formula.execute(params, executionContext as SyncExecutionContext) as any;
```

Comment:
maybe adding two method signatures instead of doing the type casting here. at least it prevents returning unexpected types. 

```
async function doFindAndExecutePackFunction<T extends FormulaSpecification>({
  params,
  formulaSpec,
  manifest,
  executionContext,
  updates,
}: FindAndExecutionPackFunctionArgs<T>): Promise<PackFunctionResponse<T>>;
async function doFindAndExecutePackFunction<T extends FormulaSpecification>({
  params,
  formulaSpec,
  manifest,
  executionContext,
  updates,
}: FindAndExecutionPackFunctionArgs<T>): Promise<
  GenericSyncFormulaResult | GenericSyncUpdateResultMarshaled | PackFormulaResult
>
```

Reply:
Yeah overloads seem like a good way to preserve the strong typing amid all the generics.

Reply:
Cool.